rfc-156-auto-merge-internal-prs.md

```diff
@@ -110,93 +110,81 @@ Therefore, we propose writing an additional `validate_not_a_bank_holiday` step i
-          - rubocop-govuk
-        merge-versions:
-          - patch
-          - minor
-```
+Defining where the service is hosted, and how the keys are stored and accessed, are out of scope for this RFC.
 
-[creating-actions]: https://docs.github.com/en/actions/creating-actions/about-custom-actions
+[govuk-saas-config-webhooks]: https://github.com/alphagov/govuk-saas-config/blob/8cfb23db6a200214411fc1de150d3c8582950588/github/lib/configure_repo.rb#L55
 
-### Raise initial config PRs
+### Trial the auto-merge service on one repo
 
-- What: Raise PRs for all applicable GOV.UK repos, to enable auto-merging of certain dependencies using GitHub Actions 
-- Who: Platform Reliability
-- When: After enabling the org-wide setting
+For one repository, Platform Reliability will create the auto merge configuration file and also configure Dependabot to raise PRs during office hours. They will trial it for a number of weeks to ensure that the auto merge service works as intended.
 
-Platform Reliability intend to reference the homemade GitHub Action from [Packaging as a GitHub Action](#packaging-a-github-action), raising an initial config PR across all GOV.UK repos that represent continuously deployed apps. Whilst Platform Reliability would create the initial PRs, it would be up to the owning teams to decide whether or not to merge, and to what extent they'd want to reconfigure the repo and version scopes.
+The configuration will be expanded from the initial spike, to include both patch and minor version bumps, of the following internal dependencies (but only where those dependencies are present in the repo):
 
-Platform Reliability also intend to raise the necessary PRs to restrict Dependabot 'schedule times' as per [5. During office hours](#5-during-office-hours). As above, it will be up to the owning teams to decide whether or not to merge, though we will stipulate that they must not enable auto-merging without also restricting the Dependabot schedule times.
+- `govuk_publishing_components` (18 versions - not including 3 major increases - between October 2022 and January 2023, across 32 apps)
+- `govuk_app_config` (6 versions - not including 1 major release - between October 2022 and January 2023)
+- `gds-api-adapters` (3 major versions between October 2022 and January 2023. 0 minor/patch ones. However, between July and August 2022 there were 4 qualifying patch increases we could have automated.)
+- `rubocop-govuk` (2 versions - 0 major ones - between October 2022 and January 2023, across a large number of repositories)
 
-The PR will be expanded from the initial spike, to include both patch and minor version bumps, of the following internal dependencies (but only where those dependencies are present in the repo):
+For `govuk_publishing_components` alone, we've had 576 PRs between October 2022 and January 2023, all of which could have been auto mergeable. If we estimate a minute of due diligence per PR, that amounts to almost 10 hours of wasted dev time.
```

Comment:
Sometimes this will be considerably more than a minute (speaking as someone who has, on occasion, been waiting for PRs to appear). Removal of the context switching penalty is also useful. We possibly need to be a little careful of ["is it worth the time"](https://xkcd.com/1205/), but I think the reduction in drudgery also has value.

I wonder also about whether we should look into a bulk "invoke dependabot check" button for deploys that might be less routine than waiting for the following day.